version: '3'
services:
  pyth-client:
    build:
      context: .
      dockerfile: cmd/pyth-client/Dockerfile
    environment:
      - REDIS_PW=${REDIS_PW}
      - REDIS_ADDR=${REDIS_ADDR}
      - NETWORK_NAME=${NETWORK_NAME}
      - CONFIG_PATH=/prices_conf
    depends_on:
      - redis-timeseries
    networks:
      - app-net
    restart: on-failure
    logging:
      options:
        max-size: "10m"
        max-file: "10"
    configs:
      - prices_conf
  ws-server:
    build:
      context: .
      dockerfile: cmd/ws-server/Dockerfile
    environment:
      - REDIS_PW=${REDIS_PW}
      - REDIS_ADDR=${REDIS_ADDR}
      - NETWORK_NAME=${NETWORK_NAME}
      - CONFIG_PATH=/prices_conf
    depends_on:
      - redis-timeseries
    ports:
      - 8080:8080
    networks:
     - app-net
    restart: on-failure
    logging:
      options:
        max-size: "10m"
        max-file: "10"
    configs:
      - prices_conf

  redis-timeseries:
    image: redis/redis-stack-server:latest
    command: redis-stack-server --requirepass ${REDIS_PW}
    environment:
      - TZ=UTC
    ports:
      - "6379:6379"
    networks:
      - app-net
    restart: on-failure
    logging:
      options:
        max-size: "10m"
        max-file: "10"
configs:
  prices_conf:
    file: ./config/prices.config.json
networks:
  app-net:
